components/lib/datatable/DataTable.js

```diff
@@ -1219,7 +1221,12 @@ export const DataTable = React.forwardRef((props, ref) => {
                     let cellData = ObjectUtils.resolveFieldData(record, field);
 
                     if (cellData != null) {
-                        cellData = props.exportFunction ? props.exportFunction({ data: cellData, field, rowData: record, column }) : String(cellData).replace(/"/g, '""');
+                        if (props.exportFunction) {
+                            cellData = props.exportFunction({ data: cellData, field, rowData: record, column });
+                        } else {
+                            cellData = String(cellData).replace(/"/g, '""');
+                            cellData = String(cellData).replace(/\n/g, '\u2028');
```

Comment:
I wonder if on the second line yu need String() since its already converted to a string?

```js
cellData = String(cellData).replace(/"/g, '""');
 cellData = cellData.replace(/\n/g, '\u2028');
 ```
or can you chain them?

```js
cellData = String(cellData).replace(/"/g, '""').replace(/\n/g, '\u2028');
```
??

Reply:
nd actually shouldn't we do the same in headers with the `"` as couldn't a header have a quote in it that would need to be escaped as well?

Reply:
Yes, that's true. I can chain, I think it is better.

Reply:
> nd actually shouldn't we do the same in headers with the `"` as couldn't a header have a quote in it that would need to be escaped as well?

To be honest, I did not think about that. Can DataTable support header with newline character?

Reply:
Yeah I jus ttried it with the Showcase changing header from Category Name to `'Category " \n Name'`

Reply:
Okay so with https://github.com/primefaces/primereact/pull/4002/commits/ed5a221b7b084aed722c7679a0e8928608fef295 I think it should be fine, can you try?